src/service/save-dispense.js

```diff
@@ -47,4 +47,38 @@ const saveDispense = async (data, sourceStockRoom) => {
 
 	return postRequest(stockOperationURL, requestBody);
 };
-export default saveDispense;
+const saveDispenseForAdhocDispense = async (data, sourceStockRoom) => {
+	const instanceTypeResponse = await getRequest(
+		stockOperationTypeURL("Distribution")
+	);
+
+	const itemArray = [];
+	if (data.dispense_drugs && Array.isArray(data.dispense_drugs)) {
+		const promises = data.dispense_drugs.map(async (item) => {
```

Comment:
async not required because there is not asynchronous operation inside the map function